### ISSUE TITLE
<reserved>

### ISSUE THREAD

**Issue by zadjii-msft** [microsoft/terminal#9996]
not blank!

**Comment by dag03tsc** [comment]
> Is the monarch a phoenix (when it's killed, who restarts it?)

I'm pretty sure this already came to your mind before, but what about using the service approach?
It'd only require the tray icon feature to enable the persistence and provide some control over the service itself.

I guess that'd also allow termination commands like `exit`, while keeping the `globalSummon` or `quakeMode` available, for as long as the service is running.

Could it be useful to have some kind of a master orchestrator service? Something like a service to host and manage all WT instances?

**Comment by kensykora** [comment]
Been testing out the preview channel and I LOVE quake mode. Since the terminal is so core to my development workflow, it'd be HUGE if this launched on startup so it felt like a core part of the OS.

**Comment by hensz** [comment]
Is this issue purely about starting (and keep running) Terminal at startup? I was looking for a way to manually start a terminal in Quake mode, just as asked for in issue #10206.

**Comment by zadjii-msft** [comment]
>  I was looking for a way to manually start a terminal in Quake mode

That'd be [`wt -w _quake`](https://github.com/microsoft/terminal/issues/10206#issuecomment-848913937). See also [the docs](https://docs.microsoft.com/en-us/windows/terminal/tips-and-tricks#quake-mode)

**Comment by hensz** [comment]
This doesn't work since I have both the stable and the preview version installed, and wt runs the stable version. Most likely I could simply uninstall the stable version. Is there any other way to launch the preview?

**Comment by zadjii-msft** [comment]
> Is there any other way to launch the preview?

You could always change `wt` to open the Preview version. To do that, you can check out the "Manage app execution aliases" page in the settings

**Comment by hensz** [comment]
Hmm, if I do this and try to run wt, I get an error from the stable version:
[Window Title]
C:\Program Files\WindowsApps\Microsoft.WindowsTerminal_1.8.1521.0_x64__8wekyb3d8bbwe\wt.exe

[Content]
Windows cannot access the specified device, path, or file. You may not have the appropriate permissions to access the item.

**Comment by hensz** [comment]
OK, when I remove the stable version running wt successfully launches the preview, and also launching in Quake mode works fine. Thanks a lot! Now I only need to figure out how I can add this command as a shortcut to my start menu 🙈

**Comment by NinovanderMark** [comment]
If there were an option to start Windows Terminal in Quake mode, but hidden, that would solve it in my book. 

Something like this:
```wt -w _quake --hidden```

**Comment by XQDD** [comment]
> If there were an option to start Windows Terminal in Quake mode, but hidden, that would solve it in my book.
> Something like this: `wt -w _quake --hidden`

it does not need boot  at startup with hidden, you can use it combine with [PowerToys](https://github.com/microsoft/PowerToys)

The `keyboard manager`'s setting:
![image](https://user-images.githubusercontent.com/20871761/163901900-8e9c4958-969a-480b-b058-e3aba2b61cbb.png)

**Comment by zadjii-msft** [comment]
Another idea suggested in #13210 - start Terminal minimized to the tray on system startup. 

I'm not sure how I feel about that. Initial thoughts is that I'd want to avoid that solution, since it would incur starting the user's shell(s) at startup, and having them running until the user actually invokes the Terminal. _(I should be OOF this week so I'm not gonna give this much more thought ATM beyond just de-duping threads)_

**Comment by vadimkantorov** [comment]
This minimized to tray on startup is quite frequent among other apps. If needed, spawning actual shell / profile can be postponed until the first summon (especially for WSLv2 shells that spawn a virtual machine and occupy a lot of RAM), but the rest of the app would already be initialized.

And in quake scenario, having Terminal occupying space in the app bar is wasteful, having one in tray is sufficient.

**Comment by PathogenDavid** [comment]
Unless there's other useful functionality associated with it, I'd personally rather not have yet another thing cluttering the overflow menu of the notification tray (and definitely not in the taskbar.)

To me this feature should feel like an ambient operating system thing.

**Comment by JJRcop** [comment]
For the moment launching minimized to tray could be solved with a command line option "--minimized"

**Comment by AlexHunterCodes** [comment]
"Launch on machine startup" and "Always display an icon in the notification area" both enabled achieves _similar_ behaviour to what I'm used to with always-available Quake mode terminals on macOS and Linux, but it's not perfect.

I have to accept the always-present tray icon that does nothing and is either visible or not visible when a normal Terminal window is open (depending on how you configure it), and I have to manually close the default Terminal window that opens on startup every time (although I'll try @JJRcop's suggestion to suppress that). Plus, when I open the Quake terminal, it appears as another Terminal window in the taskbar.

Ideally, there would be an option to launch into the background without any window at startup, **and** an option to have Terminal running in the background but without yet another tray icon. Anyone using Quake mode is a developer or a power user, if Terminal breaks or needs restarting we know how to kill it from Task Manager without a tray icon adding clutter.

Definitely like @PathogenDavid's perspective that if Quake mode is enabled, it should just _be available_ whenever without jumping through a sequence of obscure configuration options and workarounds. I know some Linux desktop environments do that already and it's really nice.

**Comment by JJRcop** [comment]
> (although I'll try @JJRcop's suggestion to suppress that)

Sorry, allow me to clarify that `--minimized` does not currently exist, and my suggestion is directed towards near-future quick-fix development rather than towards current usage.

My sincerest apologies, sorry for any misunderstanding. 😅

**Comment by zadjii-msft** [comment]
FWIW, `--minimized` is tracked in #7374

**Comment by akovia** [comment]
Probably missing something, but I don't see a reason to have anything start with the OS unless there is some large configuration to load that took an inordinate amount of time. When you hit your launcher Hotkey, it should just launch directly into quake mode as the title suggests. Having a taskbar icon doesn't bother me at all as it tucks away in the overflow and just gives another way to interact with the program to access settings or kill the program for people who are used to working that way. There really doesn't need to be an icon in the app bar though when using quake mode. Having quake mode enabled pretty much insinuates you will be accessing your terminal via hotkey, so an app bar icon should at least be configurable if it's not desired.

Linux has perfected most of these issues over the years. I use XFCE4-terminal and it does almost everything perfectly. After booting is finished I just hit F12 and it launches nearly instantly. It has a tray icon but not an app icon. If I launch any additional instances it opens in a regular window with an app bar icon, but it doesn't affect my quake mode instance. I can also launch programs directly into a new tab of my existing running quake mode instance which is extremely handy for my custom actions in Thunar.

I only use Windows at work but I am truly grateful to have WT with quake mode now. If there is one thing I actually prefer with WT, is the ability to open with a pre defined set of tabs. (though the tabs should really be on the bottom of the window imho, ([and visible](https://github.com/microsoft/terminal/issues/11085#issuecomment-999189691)) ) 

Lastly, it may be trivial, but I do wish the window would slide down from the top instead of opening across the screen from the bottom.

**Comment by DHowett** [comment]
> When you hit your launcher Hotkey, it should just launch directly into quake mode as the title suggests.

Unfortunately, the problem is that we can't do that without having something launched _before you press the hotkey_ to register for it and intercept it when you press it.

The only other way for us to accomplish it without pre-launching something would be to add a hook in the Windows shell, which wouldn't come out to users for another 1,000 years and then would only be available to people with Intel Core i48 processors.

**Comment by akovia** [comment]
> Unfortunately, the problem is that we can't do that without having something launched _before you press the hotkey_ to register for it and intercept it when you press it.

Thank you for the reply and sorry for my ignorance, but couldn't you use a systemwide hotkey to launch wt with the options you wanted? (ie.. create a shortcut and assign it a "shortcut key"?) If it determines the program is already open, it would function as a visibility toggle, otherwise it would spawn the program itself. It wouldn't be a hotkey that you configure within wt itself, but not sure that would matter if it accomplished the same goal.

I admit my ignorance on the inner workings of Windows, so again I apologize if there is something fundamentally flawed with this approach. It is completely useable as is and I'm glad to have it. Thanks for your hard work!

**Comment by DHowett** [comment]
HUH. Now that you mention it . . . shortcuts _do_ support that! I wonder if they need to be placed in a specific directory, or need some other special treatment.

That's an excellent idea. Thank you!

**Comment by PathogenDavid** [comment]
> I wonder if they need to be placed in a specific directory

I thought they could be anywhere, but it seems they must be on the desktop or a folder on the desktop. (Maybe other places too, but it's definitely not anywhere.)

It's worth noting it doesn't support shortcuts involving the Windows key (at least the UI doesn't let you create them.) Additionally, on my PC at least there's a pretty significant/unacceptable delay between me pressing the shortcut and the application actually launching.

I kinda get the vibe that this feature exists mainly due to legacy momentum. It's been around since at least Windows 95:

![Screenshot of a Windows 95 properties window for a shortcut showing the shortcut key option.](https://user-images.githubusercontent.com/278957/178572409-1c349871-c500-444d-92d1-454653b143d9.png)

It's definitely an interesting feature from a trivia perspective, but I doesn't feel particularly well-suited for this task IMO.

**Comment by Hellzbellz123** [comment]
fun fact, the desktop shortcut doesn't work properly if you select minimized in shortcut options, it results in the wt instance being spawned off screen and the only way to get it to show up after a restart is restarting WT or toggling full-screen then using win+arrow keys to move it back onto the desktop :/ this is also related too #9053 and others mentioned their, this is a decent work around if you only intend on using quake mode, however if you have windows terminal set too attach to an already opened instance you cant use WT from anywhere but win + ` until you use above workarounds, this was tested just now using windows terminal preview on windows 11

**Comment by KKirschi** [comment]
dirty way to start it minimized: wt -w _quake powershell -nologo -window minimized

**Comment by NinovanderMark** [comment]
> dirty way to start it minimized: `wt -w _quake powershell -nologo -window minimized`

That's *exactly* what I was looking for. Thank you so much! I can't quite figure out how to pass a particular Windows Terminal profile though. I tried the following, but it pops up an error about the `-window` argument being unexpected.

`wt -w _quake -p "PowerShell" -window minimized`

**Comment by DHowett** [comment]
Ah, the ages-old issue with `-p x`. You'll need to specify a command to run _using_ that profile¹; try `wt -w _quake -p PowerShell pwsh -nologo -window minimized`


<details>

<summary>¹ why is this the case?</summary>

Passing a command on the commandline _overrides_ the profile's command, it doesn't add to it (various reasons this is true, one of which is that appending arguments to `cmd` profiles would require `cmd /s /c` and appending to powershell profiles would require `-command`, unless you wanted to pass arguments and *not* a command, at which point it's all for naught really... see docker's `ENTRYPOINT` for all the terribly fun ways this can fall apart 😄)

</details>

**Comment by zeroZshadow** [comment]
Note that this doesn't seem to apply all settings for the given profile. For example, the set Font is not applied.

**Comment by alexanderVprice** [comment]
This is a temporary workaround. It doesn't start the terminal with a global shortcut, but it "simulates" the experience to an extent by starting WT hidden in the tray at login.

1. add the block

    ```powershell
    if ($START_WT_HIDDEN-eq $true) {
        Start-Process wt.exe -WindowStyle Hidden
    } 
    ```

    to a pwsh script `startup.ps1` which starts apps hidden when possible or starts them minimized, then hides them in the tray (e.g. outlook 365)
2. use the context menu in windows explorer to create a shortcut to `startup.ps1`, go into properties and change `Run:` `Normal Window.` to `Run:` `Minimized`

3. Move the shortcut to `shell:startup`

**Comment by Potato22** [comment]
> wt -w _quake powershell -nologo -window minimized

I really wished I could somehow only run this command once since the `_quake` window always pops up then quickly disappears every time a new window is opened (aka new instance in which the profile has to be re-init)

**Comment by KKirschi** [comment]
> > wt -w _quake powershell -nologo -window minimized
> 
> I really wished I could somehow only run this command once since the `_quake` window always pops up then quickly disappears every time a new window is opened (aka new instance in which the profile has to be re-init)
> 
> ##### _I put the command in the profile._

I don't exactly know what you mean, but yeah, it pops up for a second when launched. The problem is that you could shorten this delay by removing the "-nologo", but then the subshell gets really messed up.

**Comment by batu1579** [comment]
I made a few tweaks to @KKirschi 's way, and so far it's working.

1. Created a shortcut, put the following line in "Target", and select Minimize window to run
    ```
    "C:\Program Files\PowerShell\7\pwsh.exe" -Command "wt -w _quake 'C:\Program Files\PowerShell\7\pwsh.exe'  -NoExit  -Command '& {clear}' -window minimized;Start-Sleep -Seconds 1;$app=New-Object -ComObject shell.application;$app.ToggleDesktop();"
    ```
2. Move the shortcut to `C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Startup`

I added a 1 second wait after launch quake. If the window doesn't disappear after logging in, try to extend it for a little longer.

Hope that helps orz~

**Comment by zadjii-msft** [comment]
I know that the "start the Terminal without a visible window" is a big part of this, but #14944 should make this a heck of a lot easier going forward.

**Comment by User1785604260** [comment]
Will it be possible to enforce that the headless terminal window is the only window capturing the Win+` quakeMode hotkey? I don't like having the quake terminal run as admin, so I'd want to start the headless as non-admin and make sure it's the only thing opening new terminals when Win+` is pressed.

I wrote my own script to act as an Emporer and take over the Win+` hotkey entirely on its own to ensure this wouldn't happen. Sometimes I'd end up opening an admin terminal immediately on login, and if it was the first instance to run it'd grab the Win+` hotkey and then I was locked into having admin quakeMode until I quit that initial terminal.

**Comment by zadjii-msft** [comment]
After #14944 it'll still be whoever launches first who gets it. We're vaguely tracking the elevated/unelevated wonkiness in #12173 though, so that's probably the thread to follow up in 😊

**Comment by zadjii-msft** [comment]
Since folks in this thread are probably interested: https://github.com/microsoft/terminal/releases/tag/v1.18.1421.0 added the global setting `compatibility.allowHeadless`, which should let the Terminal keep running in the background after closing all windows. This'll let the global hotkeys keep working, even if the Terminal isn't open.

```jsonc
    // Add support for running the Terminal without _any_ windows #14944
    // (you probably also want at least `quakeMode` or a `globalSummon`)
    "compatibility.allowHeadless": true,
```

There's still more work to do, and I'm tracking it as follows:
* Start the terminal minimized: #7374
  * You can probably work with `start /min wt` for most scenarios as of 1.18.
* Start the terminal "headless", without any windows: #9996
* Enable `startOnUserLogin` to be set to `headless`, so the terminal auto-starts without a window: #13630

**Comment by vadimkantorov** [comment]
@zadjii-msft could Terminal somehow run headless / without UI or with minimal non fully-fledged UWP UI if a `*.cmd` (or other) script is being launched in an unattended way as part of Windows startup:

**Comment by zadjii-msft** [comment]
@vadimkantorov I'm gonna move that discussion over to #15001.

**Comment by vadimkantorov** [comment]
no prob! just wondering if this new headless mode is useful for that usecase

**Comment by DrLeh** [comment]
I use the following hotkey in Autohotkey to help enable this:

```ahk
$#`::
If (ProcessExist("WindowsTerminal.exe"))
    Send #``
else
    Run "wt"
::
```

I couldn't get this to actually start in quake mode, so in reality what i have to do is press `#\``, this opens terminal, then press it again to get quake mode, then again to hide it, then close the original terminal. 

This is far from optimal but at least gives me a single hotkey to get terminal open and into quake mode without having to open wt through some other means.

**Comment by NewtonChutney** [comment]
> Since folks in this thread are probably interested: https://github.com/microsoft/terminal/releases/tag/v1.18.1421.0 added the global setting `compatibility.allowHeadless`, which should let the Terminal keep running in the background after closing all windows. This'll let the global hotkeys keep working, even if the Terminal isn't open.
> 
> ```jsonc
>     // Add support for running the Terminal without _any_ windows #14944
>     // (you probably also want at least `quakeMode` or a `globalSummon`)
>     "compatibility.allowHeadless": true,
> ```

Meanwhile, can the preview release of WT point to a schema with these keys in it?
I did try changing the schema, and didn't find this key though.. This seems missing.. 🫠
But the key value pair is detected! 😌

![20230703_124156.jpg](https://github.com/microsoft/terminal/assets/70827815/5e65d158-d029-4d74-bdfc-d910bbd640c0)

**Comment by Memoraike** [comment]
``> I use the following hotkey in Autohotkey to help enable this:
> 
> ```ahk
> $#`::
> If (ProcessExist("WindowsTerminal.exe"))
>     Send #``
> else
>     Run "wt"
> ::
> ```
> 
> I couldn't get this to actually start in quake mode, so in reality what i have to do is press `#``, this opens terminal, then press it again to get quake mode, then again to hide it, then close the original terminal.
> 
> This is far from optimal but at least gives me a single hotkey to get terminal open and into quake mode without having to open wt through some other means.

Try this:
```
ProcessExist(Name){
	Process,Exist,%Name%
	return Errorlevel
}

#If !ProcessExist("WindowsTerminal.exe")
#`::
Run wt -w _quake
Return
```
![image](https://github.com/microsoft/terminal/assets/46358138/e808a466-8d77-4fd8-90fd-80f286e9d235)

**Comment by H4M5TER** [comment]
> dirty way to start it minimized: `wt -w _quake powershell -nologo -window minimized`

Since it would open a new shell nested in default shell, I put `pwsh -nologo -window minimized` on a standalone wt profile and use `wt -w _quake -p quake` to launch.

**Comment by MolotovCherry** [comment]
To summarize everything in this article for people, as of my current posting date, this works for me:

As a startup task. (By default it won't run the initial profile)
`wt -w _quake powershell -nologo -window minimized -NoExit -Command "& { . 'C:\Users\><User>\Documents\PowerShell\Microsoft.PowerShell_profile.ps1' }"`

Inside your settings.json (so you can still open quake after terminal/quake windows are closed)
`"compatibility.allowHeadless": true,`

**Comment by MolotovCherry** [comment]
@unennhexium I have mine in a task schedule job.
Program `C:\Users\<user>\AppData\Local\Microsoft\WindowsApps\wt.exe`
Args `-w _quake powershell -nologo -window minimized -NoExit -Command "& { . 'C:\Users\<user>\Documents\PowerShell\Microsoft.PowerShell_profile.ps1' }"`
Start in `C:\Users\<user>`

I'm sure a shortcut way exists, but I have no need to experiment with that since it's already working for me

**Comment by amzon-ex** [comment]
> Program `C:\Users\<user>\AppData\Local\Microsoft\WindowsApps\wt.exe`
Args `-w _quake powershell -nologo -window minimized -NoExit -Command "& { . 'C:\Users\<user>\Documents\PowerShell\Microsoft.PowerShell_profile.ps1' }"`
Start in `C:\Users\<user>`

@MolotovCherry could you please tell me what `-NoExit` does?

**Comment by MolotovCherry** [comment]
> > Program `C:\Users\<user>\AppData\Local\Microsoft\WindowsApps\wt.exe`
> > Args `-w _quake powershell -nologo -window minimized -NoExit -Command "& { . 'C:\Users\<user>\Documents\PowerShell\Microsoft.PowerShell_profile.ps1' }"`
> > Start in `C:\Users\<user>`
> 
> @MolotovCherry could you please tell me what `-NoExit` does?

@amzon-ex According to [microsoft docs](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_powershell_exe?view=powershell-5.1#-noexit), it:
> Doesn't exit after running startup commands.

**Comment by MolotovCherry** [comment]
@unennhexium Your task works fine for me verbatim (except I added `-NoExit` and of course changed the user path to mine).

**Comment by MadameMinty** [comment]
In `Start Menu\Programs\StartUp`, I put a shortcut to `wt.exe -w _quake` with the dropdown `Run` set to `Minimized`. Upon startup, WT flickers once, then Quake Mode shortcut key works to show/hide the terminal until I alt+F4 or otherwise shutdown the process. It's not perfect, but close enough.

**Comment by fleps** [comment]
**I can't believe we still don't have a WT argument --minimized yet.**

The solutions proposed here do not work flawlessly, one (big) caveat for me is that you lose the ability to open your default profile Elevated / as administrator, even if configured on WT.

Can we PLEASE get a `--minimized` considering we already have `--maximized`???

**Comment by GeorgeGedox** [comment]
Could we know what is stopping the devs from implementing a proper option to start windows terminat in the background on startup if we want this?
I'd like to be able to have quake mode available without using weird startup hacks or having a terminal instance active at all times in my taskbar, that defeats the purpose of quake mode.

The whole idea behind quake mode is to have a hotkey that opens the terminal at any time from anywhere without any previous intervention to "prep" it beforehand.

**Comment by zadjii-msft** [comment]
> Could we know what is stopping the devs from implementing a proper option to start windows terminal in the background on startup if we want this?

Time, basically. I haven't had the chance to loop back around on `--headless` (as mentioned in https://github.com/microsoft/terminal/issues/9996#issuecomment-1452670063), or the "headless" startup task (as referenced in https://github.com/microsoft/terminal/issues/9996#issuecomment-1561249825 and further spec'd in https://github.com/microsoft/terminal/issues/13630#issuecomment-1452590629).

Not sure when exactly I'll have a chance to. I don't think these would be terribly difficult if someone in the community wanted to contribute either here. I could definitely provide guidance and pointers if someone runs into any questions.

**Comment by MaverickMartyn** [comment]
> **I can't believe we still don't have a WT argument --minimized yet.**
> 
> The solutions proposed here do not work flawlessly, one (big) caveat for me was losing the ability to open your default profile Elevated / as administrator, even if configured on WT.
> 
> My workaround to have it as ADMIN was similar to above:
> 
> * Go to your startup folder (run shell:startup)
> * Create a shortcut and on Target add `wt pwsh -window minimized` (I don't use quake mode, I have my Win+` set to globalSummon, you can add/keep the quake parameters if wanted)
>   
>   * In this case, make sure on your WT _Settings_ > _Appearance_ you have the option to hide the terminal to tray area when minimized
> * Set "Run" option also to Minimized (this prevents any flicker)
> * Click on Advanced button and check "Run as Administrator" (don't use the similar option from Compatibility tab, it's not the same, you want the one from the Advanced button)
> * Apply, OK.
> 
> Still, can we PLEASE get a `--minimized` considering we already have `--maximized`???

In regards to running as admin, I use PowerToys' Keyboard Manager to run terminal on CTRL+ALT+T, and CTRL+SHIFT+ALT+T runs it as administrator.
Assuming you can make WT start in quake mode with a parameter, that should work too.

**Comment by only-su** [comment]
> > **I can't believe we still don't have a WT argument --minimized yet.**
> > The solutions proposed here do not work flawlessly, one (big) caveat for me was losing the ability to open your default profile Elevated / as administrator, even if configured on WT.
> > My workaround to have it as ADMIN was similar to above:
> > 
> > * Go to your startup folder (run shell:startup)
> > * Create a shortcut and on Target add `wt pwsh -window minimized` (I don't use quake mode, I have my Win+` set to globalSummon, you can add/keep the quake parameters if wanted)
> >   
> >   * In this case, make sure on your WT _Settings_ > _Appearance_ you have the option to hide the terminal to tray area when minimized
> > * Set "Run" option also to Minimized (this prevents any flicker)
> > * Click on Advanced button and check "Run as Administrator" (don't use the similar option from Compatibility tab, it's not the same, you want the one from the Advanced button)
> > * Apply, OK.
> > 
> > Still, can we PLEASE get a `--minimized` considering we already have `--maximized`???
> 
> In regards to running as admin, I use PowerToys' Keyboard Manager to run terminal on CTRL+ALT+T, and CTRL+SHIFT+ALT+T runs it as administrator. Assuming you can make WT start in quake mode with a parameter, that should work too.

When using powertoys Keyboard Manager to run terminal is there a way to make it not open a new tab when a window already exists? In other way to make the command create a window when none exists or just invoke the existing tab instead of opening a new one whe it does?

**Comment by Memoraike** [comment]
> When using powertoys Keyboard Manager to run terminal is there a way to make it not open a new tab when a window already exists? In other way to make the command create a window when none exists or just invoke the existing tab instead of opening a new one whe it does?

For your purpose, autohotkey will work.  It doesn't work perfectly with minimized terminal windows, but as a temporary solution it's great. Don't forget to change keyboard shortcuts if necessary.

 ```ahk
#Requires AutoHotkey v2.0
#SingleInstance

DetectHiddenWindows true
terminalName := "WindowsTerminal.exe"

; [Win Shift Enter] If the terminal is already open, activate it. Otherwise, open it.
#+Enter:: {
    if (pid := ProcessExist(terminalName)) {
        if WinExist("ahk_pid" pid) {
            PostMessage 0x0112, 0xF120, , , WinGetID("ahk_pid" pid)
            WinActivate
        }
    } else {
        Run("wt")
    }
}

; [Win Enter] Open the terminal in a new window.
#Enter:: {
    Run("wt")
}

; [Win `] Open the terminal in a quake mode. Only works if the terminal is not already open. Require setting same hotkey in the terminal.
#HotIf !ProcessExist(terminalName)
#`:: {
    Run("wt -w _quake")
}
 ```

To work correctly with minimized windows, you should enable this option in the settings:

<details>

![image](https://github.com/microsoft/terminal/assets/46358138/0337456b-0c37-4d21-adb4-0f802a4763d2)
</details>

**Comment by MaverickMartyn** [comment]
For anyone on Windows 11, you can start a new Terminal using WIN+X, followed by I for Terminal or A for elevated Terminal.  
It's not perfect, but it works without any 3rd party software or setup.

**Comment by zadjii-msft** [comment]
@User1785604260 That changed in 1.18, like, two years ago. Ever since that release, all Terminal windows are the same process

(and after 1.23, they'll all be on the same thread too)

**Comment by gitclonehama** [comment]
First off, love the work done with Windows Terminal and over all the whole dev ecosystem with the introduction of WSL. I have been going down the rabbit hole of the many discussions on this topic.

Currently, we can utilize the windows hotkeys: Win + x then i to fire the Terminal application.
We can then initiate an instance in Quake Mode by using hotkeys: Win + ` (or ~). At this point, I am very happy with the overall workflow and behavior of how Terminal behaves (love that putting a quake window into fullscreen (F11) keeps it in fullscreen even after hide/unhide using Win + `.

The only thing I wasn't clear on yet, do we have an elegant solution to streamline this process to happen automatically on start up?

Currently I have the "Start terminal instance on start-up" checked and so I don't even have to go through the Win + x then i so essentially, I only have to initiate the quake mode by hitting Win + ` with the terminal opened on start-up.

I know this is now small issue to even think about since I would only have to do this once when working on my PC, but are there any plans for a native solution/ setting to make the above actions happen without the intermediary step of having a Terminal open to initiate the default Terminal instance in Quake Mode on start up?

**Comment by thwyster** [comment]
It's 2026 and we still don't have a way to start the minimized version of Quake Mode along with the operating system so we can press Win+` and open Quake Mode?

**Comment by zadjii-msft** [comment]
I left my notes in https://github.com/microsoft/terminal/issues/13630, and in https://github.com/microsoft/terminal/issues/9996#issuecomment-1561249825. Any enterprising developer who's passionate about this feature is more than welcome to open a PR for this. 

Really the hard open question is just "how does this feature interact with state restoration?" Should we:
* for `--minimized`:
  * **1A** restore all the windows, but also SW_HIDE them to minimize them? 
  * **1B** Or just restore them normally, and create the new window minimized.
* for `--headless`:
  * **2A** restore them on launch? 
  * **2B** not restore them until the first window is created?

Hell I think you could just go with 1B and 2A and call it a day.

**Comment by thwyster** [comment]
> I left my notes in [#13630](https://github.com/microsoft/terminal/issues/13630), and in [#9996 (comment)](https://github.com/microsoft/terminal/issues/9996#issuecomment-1561249825). Any enterprising developer who's passionate about this feature is more than welcome to open a PR for this.
> 
> Really the hard open question is just "how does this feature interact with state restoration?" Should we:
> 
>     * for `--minimized`:
>       
>       * **1A** restore all the windows, but also SW_HIDE them to minimize them?
>       * **1B** Or just restore them normally, and create the new window minimized.
> 
>     * for `--headless`:
>       
>       * **2A** restore them on launch?
>       * **2B** not restore them until the first window is created?
> 
> 
> Hell I think you could just go with 1B and 2A and call it a day.

Hi @zadjii-msft, thanks for the explanation earlier.

Just to clarify: if my previous comment sounded harsher than I intended, I’m sorry about that. English isn’t my first language and sometimes the tone doesn’t come out the way I mean it.

About the design questions you raised, I wanted to share a simple user perspective that might help.

Right now, WT doesn’t restore sessions after a system reboot, at least not for me.  
If I have multiple tabs open and restart Windows, WT always starts clean.

I believe that’s already the expected behavior for most users.

Because of that, option **2B** (“don’t restore anything until the first window is created”) seems like the most appropriate choice for a headless/auto‑start scenario. It matches what WT already does today and avoids hidden windows or restoring sessions the user didn’t explicitly ask for.

Just wanted to add this point of view in case it helps the discussion.

And thanks again for all the work you put into WT!
It’s an amazing project and probably the tool I use the most in my daily workflow, right next to the big VS ❤️

**Comment by zadjii-msft** [comment]
> Right now, WT doesn’t restore sessions after a system reboot, at least not for me. If I have multiple tabs open and restart Windows, WT always starts clean.
> 
> I believe that’s already the expected behavior for most users.

Do you have this setting on?

<img width="1039" height="144" alt="Image" src="https://github.com/user-attachments/assets/72e9f0a3-d6f1-4c57-a717-b8c9c3e2fc6f" />

I believe it wasn't the default for a long time. I vaguely recall plans to make it the default, but dunno if we actually did that
